middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -1695,11 +1927,258 @@ let close_one_function acc ~code_id ~external_env ~by_function_slot decl
+      in
+      let free_names_of_params_and_body =
+        Name_occurrences.remove_continuation ~continuation:return_continuation
+          (Name_occurrences.remove_continuation ~continuation:exn_continuation
+             (Name_occurrences.remove_var ~var:my_closure
+                (Name_occurrences.remove_var ~var:my_region
+                   (Name_occurrences.remove_var ~var:my_depth
+                      (List.fold_left
+                         (fun free_names param ->
+                           Name_occurrences.remove_var free_names
+                             ~var:(Bound_parameter.var param))
+                         free_names_of_body
+                         (Bound_parameters.to_list params))))))
+      in
+      let wrapper_code =
+        Code.create code_id ~params_and_body:wrapper_params_and_body
+          ~free_names_of_params_and_body ~params_arity ~param_modes
+          ~first_complex_local_param:
+            (Function_decl.first_complex_local_param decl)
+          ~result_arity:return ~result_types:Unknown
+          ~result_mode:(Function_decl.result_mode decl)
+          ~contains_no_escaping_local_allocs:
+            (match Function_decl.result_mode decl with
+            | Alloc_heap -> true
+            | Alloc_local -> true)
+          ~stub:true ~inline:Inline_attribute.Default_inline
+          ~poll_attribute:
+            (Poll_attribute.from_lambda (Function_decl.poll_attribute decl))
+          ~check:
+            (Check_attribute.from_lambda (Function_decl.check_attribute decl))
+          ~is_a_functor:(Function_decl.is_a_functor decl)
+          ~is_opaque:false ~recursive ~newer_version_of:None ~cost_metrics
+          ~inlining_arguments:(Inlining_arguments.create ~round:0)
+          ~dbg ~is_tupled ~is_my_closure_used:true ~inlining_decision
+          ~absolute_history ~relative_history ~loopify:Never_loopify
+      in
+      let main_approx =
+        let code = Code_or_metadata.create main_code in
+        let meta = Code_or_metadata.remember_only_metadata code in
+        if Flambda_features.classic_mode ()
+        then (
+          Inlining_report.record_decision_at_function_definition
+            ~absolute_history
+            ~code_metadata:(Code_or_metadata.code_metadata meta)
+            ~pass:After_closure_conversion
+            ~are_rebuilding_terms:(Are_rebuilding_terms.of_bool true)
+            inlining_decision;
+          if Function_decl_inlining_decision_type.must_be_inlined
+               inlining_decision
+          then code
+          else meta)
+        else meta
+      in
+      (* The wrapper doesn't contain any set of closures *)
+      let slot_offsets = Slot_offsets.empty in
+      ( wrapper_code,
+        Function_slot.Map.add main_function_slot main_approx by_function_slot,
+        (main_function_slot, main_code_id) :: function_code_ids,
+        Acc.add_code ~code_id:main_code_id ~code:main_code ~slot_offsets acc )
+  in
```

Comment:
Could at least some of this code be split into separate functions ? This diff alone adds ~250 lines of uninterupted code to the body of a single function, and another diff above adds 200 lines to the same function. This makes the resulting code painful to read and understand, at least for me.

It is coherent with the rest of this module so I won't fight too hard on this, but if some parts of the code could be split into its own function, I think that would help.

Reply:
I just tried to do that, I'm not sure if it's better or not, please look!

Reply:
That indeed helps in reading the diff, thanks !